### PR TITLE
Fix compatibility with android plugin 2.+

### PR DESCRIPTION
**Pull request by dehengxu**
Increase version to 0.8.5

### CODE REVIEW

File: mavenRepo/org/robobinding/aspectj-plugin/0.8.4/aspectj-plugin-0.8.4.jar.md5 [hunk 1]
```diff
@@ -1 +1 @@
-0d2d30bffed5714a3f623be3ac521e63
```
Comment (weicheng113) [inline mavenRepo/org/robobinding/aspectj-plugin/0.8.4/aspectj-plugin-0.8.4.jar.md5:1]: Shall we keep aspectj-plugin-0.8.4 unchanged? or does it have to be changed?

File: src/main/groovy/org/robobinding/plugin/AspectJCompile.java [hunk 2]
```diff
@@ -83,7 +83,13 @@ private String getSourceRoots() {
 			@SuppressWarnings("unchecked")
 			Collection<DirectoryFileTree> dirFileTrees = (Collection<DirectoryFileTree>)method.invoke(getSource(), new Object[0]);
 			for (DirectoryFileTree dirFileTree : dirFileTrees) {
-				sourceRoots.add(dirFileTree.getDir().getAbsolutePath());
+				String sourceRoot = dirFileTree.getDir().getAbsolutePath();
+				File file = new File(sourceRoot);
+				if (!file.exists()) {
+					getLogger().debug("Skip not exist sourceRoot :" + sourceRoot);
```
Comment (weicheng113) [inline src/main/groovy/org/robobinding/plugin/AspectJCompile.java:89]: We may want to display as a warning here? And shall we rewording it something like 'Skipped a non-existent source code root directory: '?

File: build.gradle [hunk 3]
```diff
@@ -2,9 +2,9 @@ apply plugin: 'groovy'
 apply plugin: 'eclipse'
 apply plugin: 'maven'
 
-sourceCompatibility = 1.5
-targetCompatibility = 1.5
-version = '0.8.4'
+sourceCompatibility = JavaVersion.VERSION_1_7
+targetCompatibility = JavaVersion.VERSION_1_7
```
Comment (weicheng113) [inline build.gradle]: Do we have to set the compatibility level to version 1.7? If we have a choice, we may want to support as old as possible in compatibility.

**Comment by weicheng113** [comment]
@xudeheng Thanks a lot for the contributions. If you have a chance, could you please have a look the questions i commented?

File: build.gradle (hunk 3 above)
Comment (dehengxu) [reply build.gradle]: All right, I could keep your origin compatibility level. I'll send my request again.

File: mavenRepo/org/robobinding/aspectj-plugin/0.8.4/aspectj-plugin-0.8.4.jar.md5 (hunk 1 above)
Comment (dehengxu) [reply mavenRepo/org/robobinding/aspectj-plugin/0.8.4/aspectj-plugin-0.8.4.jar.md5:1]: I found  my solution only work with android plugin 2.+ versions last night. I'll make more tests today.

File: build.gradle (hunk 3 above)
Comment (weicheng113) [reply build.gradle]: Thanks @xudeheng .  We will need to test if setting to VERSION_1_5 is ok.

File: mavenRepo/org/robobinding/aspectj-plugin/0.8.4/aspectj-plugin-0.8.4.jar.md5 (hunk 1 above)
Comment (weicheng113) [reply mavenRepo/org/robobinding/aspectj-plugin/0.8.4/aspectj-plugin-0.8.4.jar.md5:1]: Appreciate your effort, @xudeheng .

Comment (dehengxu) [reply mavenRepo/org/robobinding/aspectj-plugin/0.8.4/aspectj-plugin-0.8.4.jar.md5:1]: I'm afraid cannot fix the issue cross plugin 1.+ and plugin2.+ this time.   :-(
0.8.4 and older version for less plugin 2.+
0.8.5 and newer version for great equal  plugin 2.+.

Comment (weicheng113) [reply mavenRepo/org/robobinding/aspectj-plugin/0.8.4/aspectj-plugin-0.8.4.jar.md5:1]: @xudeheng I think that is good enough. People could either choose 0.8.4 plus old Android studio, or 0.8.5+ plus Android Studio 2.+.
So in this way, we just keep aspectj-plugin 0.8.4 untouched for old versions of Android Studio? It is like we publish these artifacts into the maven central, we cannot(or don't want to) change already published artifacts.

**Comment by weicheng113** [comment]
@xudeheng I think we have a misunderstanding. Sorry, if i confused you here. I meant we keep the old published aspectj-plugin 0.8.4 unchanged. And we append the new aspectj-plugin 0.8.5 as you implemented. Thanks. You are in the RoboBinding QQ Chinese discussion group? If so, we can communicate further there if you want.

File: mavenRepo/org/robobinding/aspectj-plugin/0.8.4/aspectj-plugin-0.8.4.jar.md5 (hunk 1 above)
Comment (dehengxu) [reply mavenRepo/org/robobinding/aspectj-plugin/0.8.4/aspectj-plugin-0.8.4.jar.md5:1]: Yeah, I don't wanna change 0.8.4 anything. Maybe my words make this complicated.

0.8.5 only works well with android plugin 2.+. Do your prefer this feature?

Comment (weicheng113) [reply mavenRepo/org/robobinding/aspectj-plugin/0.8.4/aspectj-plugin-0.8.4.jar.md5:1]: Yes. I think that is also most people want. Thanks.

Comment (weicheng113) [reply mavenRepo/org/robobinding/aspectj-plugin/0.8.4/aspectj-plugin-0.8.4.jar.md5:1]: @xudeheng , Is the new 0.8.5 ready? I think the contribution is very helpful for other people. Please let me know when you are ready.

Comment (dehengxu) [reply mavenRepo/org/robobinding/aspectj-plugin/0.8.4/aspectj-plugin-0.8.4.jar.md5:1]: yeah, It's ready.

**Comment by weicheng113** [comment]
@xudeheng , Thanks for the contributions. I have migrated the fix. If there is any problem, please let me know.

Cheers, Cheng.

**Pull request closed without merging**